yktracec/src/promote.rs

```diff
@@ -0,0 +1,108 @@
+//! Software constant recorder.
+//!
+//! This is used for promoting variables to constants when compiling a trace.
+
+use std::{
+    cell::RefCell,
+    collections::{HashMap, VecDeque},
+    ffi::{c_char, CStr},
+};
+
+thread_local! {
```

Comment:
I wonder if this should, in the medium term, go into `MTThread`? That way we'd only have a single thread-local knocking around. Doesn't need to be done in this PR, though.

Reply:
From memory, I failed to put this where I wanted due to circular crate dependencies.